### DISCUSSION TITLE
The objects that I deleted is still showing in the object list

### DISCUSSION THREAD

**Discussion by SK-mechcoder** [ansys/pyaedt#5498] [General]
Hello, 

During the execution of the script, I occasionally switch to the Electronics Desktop UI to manually delete some 3D CAD objects, and then return to the script console to continue execution. In my script, I have a for loop that attempts to delete any objects that have been removed from the UI, by checking for "None" when accessing the imported bodies by id. However, even though I get a warning stating there are no objects to delete, the script continues to show the deleted objects when I print the imported_bodies list. How can I resolve this issue?
```
from ansys.aedt.core import Maxwell3d
from ansys.aedt.core.application.job_manager import get_hpc_info, update_simulation_engines, update_simulation_cores


#creates the new project
relay = Maxwell3d(project="SR6_relay", non_graphical=False, new_desktop=True)
temp_directory=r"C:\To_be_copied\automation\aedt\ProgramsforMaxwell"
relay.set_temporary_directory(temp_directory)
#renames the design
Iteration_1=relay.rename_design("Iteration_1")
#imports the model as .stp file
m3d=relay.import_3d_cad(input("give the path for importing the stp file"),create_group=False)
imported_bodies=relay.modeler.model_objects
for i in imported_bodies:
    if 'coil' in i.lower() or 'spule' in i.lower():
        obj=relay.modeler.get_object_from_name(assignment=i)
        obj.name="coil"
    elif 'yoke' in i.lower() or 'armature' in i.lower() or 'iron' in i.lower() or 'core' in i.lower():
        obj = relay.modeler.get_object_from_name(assignment=i)
        obj.name = i+"iron"
    else:
        print("no changes")
imported_bodies=relay.modeler.model_objects
print(imported_bodies)
#Material import
relay.materials.exists_material("vacuum")
magnetic_material="RFE80"
copper_material="copper"
material_library=relay.personallib
relay.materials.import_materials_from_file(input_file=f"{material_library}\RFE80.amat")
relay.materials.import_materials_from_file(input_file=f"{material_library}\copper.amat")
print(relay.materials.get_used_project_material_names())
#material assignment
for i in imported_bodies:
    if 'coil' in i.lower():
        relay.assign_material(assignment=i,material=copper_material)
    elif 'iron' in i.lower():
        relay.assign_material(assignment=i,material=magnetic_material)
    else:
        print(" good to go")
# finding the model center
extremes=relay.modeler.get_model_bounding_box()
model_center=(extremes[0]+extremes[3])/2, (extremes[1]+extremes[4])/2,(extremes[2]+extremes[5])/2
print(extremes)
print(model_center)
# translating to align to global center to model center
translation_vector=[-model_center[0],-model_center[1],-model_center[2]]
print(translation_vector)
relay.modeler.move(assignment=imported_bodies,vector=translation_vector)
print(relay.modeler.get_model_bounding_box())
##
##
input("delete the un-necessary model objects from the project, after deleting press enter to continue")

#Exceptions to handle the deleted objects

for i in imported_bodies:
    nilamai = (relay.modeler.get_obj_id(assignment=i))
    print(nilamai)
    if nilamai == None:
        relay.modeler.delete(assignment=i)
    else:
        print("good to go")
print(relay.modeler.model_objects.remove(i))
print(imported_bodies)
```

here is the messages from console.

> C:\Users\TE544300\python_venv_automation\Scripts\python.exe C:\To_be_copied\automation\aedt\ProgramsforMaxwell\ms.py 
> PyAEDT INFO: Python version 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
> PyAEDT INFO: PyAEDT version 0.11.5.
> PyAEDT INFO: Initializing new Desktop session.
> PyAEDT INFO: Log on console is enabled.
> PyAEDT INFO: Log on file C:\Users\TE544300\AppData\Local\Temp\pyaedt_TE544300_9cc10359-d38e-432f-9d96-a4ba6276a354.log is enabled.
> PyAEDT INFO: Log on AEDT is enabled.
> PyAEDT INFO: Debug logger is disabled. PyAEDT methods will not be logged.
> PyAEDT INFO: Launching PyAEDT with gRPC plugin.
> PyAEDT INFO: New AEDT session is starting on gRPC port 61342
> PyAEDT INFO: AEDT installation Path C:\ENGapps\AnsysEM\v232\Win64
> PyAEDT INFO: Ansoft.ElectronicsDesktop.2023.2 version started with process ID 1876.
> PyAEDT INFO: Project SR6_relay has been created.
> PyAEDT INFO: No design is present. Inserting a new design.
> PyAEDT INFO: Added design 'Maxwell 3D_G4M' of type Maxwell 3D.
> PyAEDT INFO: Aedt Objects correctly read
> give the path for importing the stp fileSR6.stp
> PyAEDT INFO: Modeler class has been initialized! Elapsed time: 0m 0sec
> PyAEDT INFO: Step file SR6.stp imported
> ['15587072_SR6II_ARMATURE_CALIBiron', '15587072_SR6II_ARMATURE_CALIB_1iron', '15587062_SR6II_CORE_CALIB_1_1iron', '15587052_SR6II_YOKE_CALIB_1_1iron', 'coil']
> PyAEDT INFO: Materials class has been initialized! Elapsed time: 0m 0sec
> PyAEDT INFO: Parsing C:\Users\TE544300\OneDrive - TE Connectivity\Documents\Ansoft\PersonalLib\RFE80.amat.
> PyAEDT INFO: File C:\Users\TE544300\OneDrive - TE Connectivity\Documents\Ansoft\PersonalLib\RFE80.amat correctly loaded. Elapsed time: 0m 0sec
> PyAEDT INFO: Adding new material to the Project Library: RFE80
> PyAEDT INFO: Parsing C:\Users\TE544300\OneDrive - TE Connectivity\Documents\Ansoft\PersonalLib\copper.amat.
> PyAEDT INFO: File C:\Users\TE544300\OneDrive - TE Connectivity\Documents\Ansoft\PersonalLib\copper.amat correctly loaded. Elapsed time: 0m 0sec
> PyAEDT INFO: Adding new material to the Project Library: copper
> []
> [-0.750000000000001, -6.8, -18.85, 13.449, 6.9, 2.41387920510076]
> (6.349499999999999, 0.050000000000000266, -8.218060397449621)
> [-6.349499999999999, -0.050000000000000266, 8.218060397449621]
> [-7.0995, -6.85, -10.6319396025504, 7.0995, 6.85, 10.6319396025504]
> delete the un-necessary model objects from the project, after deleting press enter to continue
> 6
> good to go
> None
> PyAEDT WARNING: No objects to delete
> 480
> good to go
> 874
> good to go
> 1279
> good to go
> None
> ['15587072_SR6II_ARMATURE_CALIBiron', '15587072_SR6II_ARMATURE_CALIB_1iron', '15587062_SR6II_CORE_CALIB_1_1iron', '15587052_SR6II_YOKE_CALIB_1_1iron', 'coil']

**Comment by Samuelopez-ansys** [comment]
@SK-mechcoder We are trying to improve this behavior, because until now, pyaedt modeler was retriving the information from the AEDT file, so it means that if you delete an object manually, the PyAEDT object is not anymore syncrhonize with AEDT modeler.

We are working a new Pull request which will fix this issue from 2024R2, if you use previous versions, you will not have this advantage.

The only workaround, is saving the project and reconnecting again:

relay.save_project()
relay.release_desktop(False, False)
relay= Maxwell3d()
